### ISSUE TITLE
[Bug]: Deserialization of DefaultClassifier model trained with loss weights fails

### ISSUE THREAD

**Issue by alanakbik** [flairNLP/flair#3585]
### Describe the bug

A DefaultClassifier if trained with special loss weights train normally, but the model cannot be loaded.

### To Reproduce

```python
from flair.datasets import CONLL_03
from flair.embeddings import TransformerWordEmbeddings
from flair.models import TokenClassifier
from flair.trainers import ModelTrainer

corpus = CONLL_03().downsample(0.01)

dictionary = corpus.make_label_dictionary("ner")

classifier = TokenClassifier(
    embeddings=TransformerWordEmbeddings("distilbert-base-uncased", fine_tune=False),
    label_dictionary=dictionary,
    label_type="ner",
    loss_weights={"B-PER": 0.0, "I-PER": 0.0, "E-PER": 0.0, "S-PER": 0.0},
)

trainer = ModelTrainer(classifier, corpus)

trainer.train("resources/models/tune", max_epochs=1)
```

### Expected behavior

Correct loading of model

##### Flair
0.14.0
##### Pytorch
2.0.0+cu117
##### Transformers
4.39.1
#### GPU
True
